test/controllers/communicator.js

```diff
@@ -1,30 +1,35 @@
 const request = require('supertest');
 const chai = require('chai');
 
-const server = require('../../app');
+const Communicator = require('../../api/models/Communicator');
+
 const helper = require('../helper');
 
 //Parent block
 describe('Communicator API calls', function () {
+  let server;
   let user;
 
   before(async function () {
-    await helper.deleteMochaUser();
-    user = await helper.prepareUser(server, {
-      role: 'user',
-      email: helper.userData.email,
-    });
+    helper.prepareNodemailerMock(); //enable mockery and replace nodemailer with nodemailerMock
+    server = require('../../app'); //register mocks before require the original dependency
   });
 
   after(async function () {
-    await helper.deleteMochaUser();
+    helper.prepareNodemailerMock(true);
+    await helper.deleteMochaUsers();
+    await Communicator.deleteMany({ author: 'cboard mocha test' });
   });
 
-  describe('POST /communicator create Communicator', function () {
-    after(async function () {
-      await helper.deleteCommunicatorById(this.communicatorid);
+  beforeEach(async function () {
+    helper.communicatorData.email = helper.generateEmail();
```

Comment:
This would be my preferred way of doing it:

```javascript
let email;
// ...
email = helper.generateEmail();
user = await helper.prepareUser(server, { role: 'user', email });
// ...
const res = await request(server)        
  .post('/communicator')
  .send({ ...helper.communicatorData, email })
  .expect('Content-Type', /json/)
  .expect(200);
```

But if the tests are working, let's leave it like that for now.

Reply:
Fine!! I agree! thanks for your review.